katsdptelstate/telescope_state.py

```diff
@@ -744,6 +744,16 @@ def keys(self, filter='*'):
         keys = self._backend.keys(_ensure_binary(filter))
         return sorted(_ensure_str(key) for key in keys)
 
+    def _ipython_key_completions_(self):
+        """List of keys used in IPython (version >= 5) tab completion.
+
+        This respects the prefix list and presents keys with prefixes removed.
+        """
+        keys = []
+        for prefix in self._prefixes:
+            keys.extend(k[len(prefix):] for k in self.keys(prefix + b'*'))
```

Comment:
This will go wrong if a prefix contains any character that's special to the globbing done by `keys`. It's also not ideal to have to fetch a key list from the remote server for each prefix - on a link to site that might be slow enough to be noticeable. I'd suggest just calling `self.keys()` once (with no filter) into a local temporary, and then doing your own filtering inside the loop.

Reply:
Very good idea, since low latency is important for interactive use - will do.

Reply:
It will even go wrong if a key contains e.g. `*` (not that it should...), because redis won't restrict the valid character set and expects the user to escape any special characters in the KEYS command. [Speaking of KEYS, I notice that SCAN is preferred these days, although I guess our database is still considered small...]